Update branching info for Rancher 2.8 (#1201)
rancher/terraform-provider-rancher2: README.md

Branching the Provider
---------------------------

> The provider is branched into two release lines that have major version alignment with Rancher 2.6, 2.7 and 2.8. The `release/v2` branch with 2.0.0+ is aligned with Rancher 2.6, the `release/v3` branch with 3.0.0+ is aligned with Rancher 2.7, and `master` with 4.0.0+ is aligned with Rancher 2.8. Terraform provider fixes and new features will be available on `master` but only critical bug fixes will be backported to `release/v3` and `release/v2`.

Reviewer: ```suggestion
The provider is branched into three release lines with major version alignment with Rancher 2.6, 2.7, and 2.8. The `release/v2` branch with 2.0.0+ is aligned with Rancher 2.6, the `release/v3` branch with 3.0.0+ is aligned with Rancher 2.7, and the `master` branch with 4.0.0+ is aligned with Rancher 2.8. Terraform provider fixes and new features will be available on `master`, but only critical bug fixes will be backported to `release/v3` and `release/v2`.
```
